File: plugins/de.ovgu.featureide.examples/featureide_examples/Email-FH-JML/features/Verify/EmailSystem/Email.java
package EmailSystem;

public class Email {
	protected boolean isSignatureVerified;
	
	static void printMail(Email msg) {
		original(msg);
		Util.prompt("SIGNATURE VERIFIED " + msg.isSignatureVerified());
	}

	boolean isSignatureVerified() {
		return isSignatureVerified;
	}

	void setIsSignatureVerified(boolean value) {
		this.isSignatureVerified = value;
	}
}

File: plugins/de.ovgu.featureide.examples/featureide_examples/EmailSystem-FH-C/features/Verify/Email.c
void
printMail (struct email *msg)
{
  original (msg);
  printf ("SIGNATURE VERIFIED\n  %i\n", msg->isSignatureVerified);
}

struct email *
cloneEmail (struct email *msg)
{
  struct email *clone = original (msg);
  clone->isSignatureVerified = msg->isSignatureVerified;
  return clone;
}

int
isVerified (struct email *msg)
{
  return msg->isSignatureVerified;
}
